### PR TITLE
Prevent hash map iterations

### PR DESCRIPTION
**Pull request by GalKauffman**

### CODE REVIEW

File: clippy_lints/src/prevent_hash_map_iterations.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                    let receiver_type = remove_module_name(remove_brackets(receiver_full_type.as_str()));
+                    let function_name = seg.ident.name.as_str();
+
+                    for (bad_type, bad_function_name) in TYPES.iter().cartesian_product(FUNCTIONS.iter()) {
+                        if *bad_type == receiver_type && *bad_function_name == function_name {
+                            span_lint_and_note(
+                                context,
+                                PREVENT_HASH_MAP_ITERATIONS,
+                                expression.span,
+                                "iterating over hashmaps is indeterministic and unreproducible, use BTreeMap instead",
```
Comment (nadavkluger) [inline clippy_lints/src/prevent_hash_map_iterations.rs:58]: non-deterministic

Comment (nadavkluger) [reply clippy_lints/src/prevent_hash_map_iterations.rs:58]: Capitalize

File: clippy_lints/src/lib.rs [hunk 2]
```diff
@@ -262,6 +262,7 @@ mod pass_by_ref_or_value;
 mod pattern_type_mismatch;
 mod permissions_set_readonly_false;
 mod precedence;
+mod prevent_hash_map_iterations;
```
Comment (nadavkluger) [inline clippy_lints/src/lib.rs:265]: Remove `prevent`, like all other detection in code

File: clippy_lints/src/prevent_hash_map_iterations.rs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+
+use clippy_utils::diagnostics::span_lint_and_note;
+
+use itertools::Itertools;
+
+declare_clippy_lint! {
+    /// ### What it does
+    ///
```
[the remark this answers is not in the document]
Comment (GalKauffman) [reply clippy_lints/src/prevent_hash_map_iterations.rs:11]: i cant

File: clippy_lints/src/prevent_hash_map_iterations.rs (hunk 1 above)
Comment (nadavkluger) [reply clippy_lints/src/prevent_hash_map_iterations.rs:58]: not reproducible

**Review by nadavkluger — APPROVED** [review]

**Pull request merged**